pkg/yurthub/filter/manager/manager.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"net"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+
+	"github.com/openyurtio/openyurt/cmd/yurthub/app/options"
+	"github.com/openyurtio/openyurt/pkg/yurthub/cachemanager"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/discardcloudservice"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/initializer"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/masterservice"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/servicetopology"
+	"github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/serializer"
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+	yurtinformers "github.com/openyurtio/yurt-app-manager-api/pkg/yurtappmanager/client/informers/externalversions"
+)
+
+type Manager struct {
+	filter.Approver
+	NameToFilterRunner map[string]filter.Runner
+}
+
+func NewFilterManager(options *options.YurtHubOptions,
+	sharedFactory informers.SharedInformerFactory,
+	yurtSharedFactory yurtinformers.SharedInformerFactory,
+	serializerManager *serializer.SerializerManager,
+	storageWrapper cachemanager.StorageWrapper,
+	apiserverAddr string) (*Manager, error) {
+	if !options.EnableResourceFilter {
+		return nil, nil
+	}
+
+	if options.WorkingMode == string(util.WorkingModeCloud) {
+		options.DisabledResourceFilters = append(options.DisabledResourceFilters, filter.DisabledInCloudMode...)
+	}
+	filters := filter.NewFilters(options.DisabledResourceFilters)
+	registerAllFilters(filters)
+
+	mutatedMasterServiceHost, mutatedMasterServicePort, _ := net.SplitHostPort(apiserverAddr)
+	if options.AccessServerThroughHub {
+		mutatedMasterServicePort = options.YurtHubProxySecurePort
+		if options.EnableDummyIf {
+			mutatedMasterServiceHost = options.HubAgentDummyIfIP
+		} else {
+			mutatedMasterServiceHost = options.YurtHubProxyHost
+		}
+	}
+
+	runnerChain, err := createRunnerChain(filters, sharedFactory, yurtSharedFactory, serializerManager, storageWrapper, util.WorkingMode(options.WorkingMode), options.NodeName, mutatedMasterServiceHost, mutatedMasterServicePort)
```

Comment:
How about calling it `createFilterRunners` instead of `createRunnerChain`, as these returned runners will be called on demand rather than in a specific order indicated by this chain. It's a little confused when reading this code.

Reply:
it's a good idea. fixed